src/logger/httpLoggerRules_test.go

```diff
@@ -286,195 +286,662 @@ func TestUsesRemoveRules(t *testing.T) {
+	helper := GetTestHelper()
+
+	request := helper.MockRequestWithJson2()
+	mockResponse := helper.MockResponseWithHtml()
+
+	request.Body = ioutil.NopCloser(bytes.NewBufferString(helper.mockJson))
+	mockResponse.Body = ioutil.NopCloser(bytes.NewBufferString(helper.mockHtml))
+	mockResponse.Request = request
+
+	queue := make([]string, 0)
+	logger := NewHttpLoggerQueueRules(queue, "!response_body! replace !blahblahblah!, !ZZZZZ!")
+	httpMessage.sendNetHttpRequestResponseMessage(logger, mockResponse, 0, 0)
+	assert.Equal(t, 1, len(queue), "queue length is not 1")
+	assert.Equal(t, true, strings.Contains(queue[0], "World"), "queue was altered unexpectedly") //default mock response should contain "World"
+	assert.Equal(t, false, strings.Contains(queue[0], "ZZZZZ"), "queue was altered unexpectedly")
+
+	queue = make([]string, 0)
+	logger = NewHttpLoggerQueueRules(queue, "!response_body! replace !World!, !Mundo!")
+	httpMessage.sendNetHttpRequestResponseMessage(logger, mockResponse, 0, 0)
+	assert.Equal(t, 1, len(queue), "queue length is not 1")
+	assert.Equal(t, true, strings.Contains(queue[0], "[\"response_body\",\"<html>Hello Mundo!</html>\"],"), "queue was not altered")
+
+	queue = make([]string, 0)
+	logger = NewHttpLoggerQueueRules(queue, "!request_body|response_body! replace !^.*!, !ZZZZZ!")
+	httpMessage.sendNetHttpRequestResponseMessage(logger, mockResponse, 0, 0)
+	assert.Equal(t, 1, len(queue), "queue length is not 1")
+	assert.Equal(t, true, strings.Contains(queue[0], "[\"request_body\",\"ZZZZZ\"],"), "queue was not altered")
+	assert.Equal(t, true, strings.Contains(queue[0], "[\"response_body\",\"ZZZZZ\"],"), "queue was not altered")
+
+	queue = make([]string, 0)
+	logger = NewHttpLoggerQueueRules(queue, "!request_body! replace !^.*!, !QQ!\n!response_body! replace !^.*!, !SS!")
+	httpMessage.sendNetHttpRequestResponseMessage(logger, mockResponse, 0, 0)
+	assert.Equal(t, 1, len(queue), "queue length is not 1")
+	assert.Equal(t, true, strings.Contains(queue[0], "[\"request_body\",\"QQ\"],"), "queue was not altered")
+	assert.Equal(t, true, strings.Contains(queue[0], "[\"response_body\",\"SS\"],"), "queue was not altered")
+
+	queue = make([]string, 0)
+	logger = NewHttpLoggerQueueRules(queue, "!response_body! replace !World!, !!")
+	httpMessage.sendNetHttpRequestResponseMessage(logger, mockResponse, 0, 0)
+	assert.Equal(t, 1, len(queue), "queue length is not 1")
+	assert.Equal(t, true, strings.Contains(queue[0], "[\"response_body\",\"<html>Hello !</html>\"],"), "queue was not altered")
+
+	queue = make([]string, 0)
+	logger = NewHttpLoggerQueueRules(queue, "!response_body! replace !.*!, !!")
+	httpMessage.sendNetHttpRequestResponseMessage(logger, mockResponse, 0, 0)
+	assert.Equal(t, 1, len(queue), "queue length is not 1")
+	assert.Equal(t, false, strings.Contains(queue[0], "[\"response_body\","), "queue was not altered")
+
+	request = helper.MockRequestWithJson2()
+	mockResponse = helper.MockResponseWithHtml()
```

Comment:
I don't think you should need ot remake the request and response, you can probably just change the request and response bodies.

Sound right?

Reply:
may be able to do `mockResponse.Request.body = ...`